pkg/manifests/manifests.go

```diff
@@ -2675,6 +2677,22 @@ func (f *Factory) ThanosQuerierService() (*v1.Service, error) {
 	return s, nil
 }
 
+func (f *Factory) ThanosQuerierPrometheusRule() (*monv1.PrometheusRule, error) {
+	return f.NewPrometheusRule(MustAssetReader(ThanosQuerierPrometheusRule))
+}
+
+func (f *Factory) ThanosQuerierServiceMonitor() (*monv1.ServiceMonitor, error) {
+	sm, err := f.NewServiceMonitor(MustAssetReader(ThanosQuerierServiceMonitor))
+	if err != nil {
+		return nil, err
+	}
+
+	sm.Spec.Endpoints[0].TLSConfig.ServerName = fmt.Sprintf("thanos-querier.%s.svc", f.namespace)
```

Comment:
nit: can we either
1) iterate over endpoints end match by name or
2) define a const name which semantically explains which endpoint is referenced here

Reply:
Done.

BTW what's the reason for not having `TLSConfig.ServerName` already set in the generated assets? Is it only because one may change the user workload namespace to something different than the default `openshift-user-workload-monitoring`?